File: 3210-find-the-encrypted-string/3210-find-the-encrypted-string.py
class Solution:
    def rev(self, s: list, i: int, j: int):
        while i < j:
            s[i], s[j] = s[j], s[i]
            i += 1
            j -= 1

    def getEncryptedString(self, s: str, k: int) -> str:
        n = len(s)
        k = k % n
        s = list(s)
        self.rev(s, 0, k - 1)
        self.rev(s, k, n - 1)
        self.rev(s, 0, n - 1)
        return ''.join(s)

File: 3210-find-the-encrypted-string/3210-find-the-encrypted-string.cpp
class Solution {
public:
    void rev(string &s, int i, int j) {
        while (i <= j) {
            swap(s[i], s[j]);
            i++;
            j--;
        }
    }
    string getEncryptedString(string s, int k) {
        int n = s.length();
        k = k % n;
        rev(s, 0, k-1);
        rev(s, k, n-1);
        rev(s, 0, n-1);
        return s;
    }
};